napari/_qt/widgets/qt_viewer_buttons.py

```diff
@@ -26,6 +28,64 @@
     from ...viewer import Viewer
 
 
+def _spawn_new_image_widget(viewer: 'Viewer') -> None:
+    """
+    Creates a dialog to add an image to viewer.
+
+
+    Parameters
+    ----------
+    viewer : napari.components.ViewerModel
+        Napari viewer containing the rendered scene, layers, and controls.
+    """
+
+    # Define an enum for array type selection
+    class BackingData(Enum):
+        NumPy = 'NumPy'
+        Zarr = 'Zarr'
+
+    # Define the magicgui widget for parameter harvesting
+    @magicgui(
+        call_button="Create",
+        name={'tooltip': "If blank, a name will be generated"},
+        dimensions={'tooltip': "The size of the new image"},
+        array_type={'tooltip': "The backing data array implementation"},
+        fill_value={'tooltip': "Starting value for all pixels"},
+    )
+    def _new_image_widget(
+        name: str = "",
+        dimensions: List[int] = [512, 512],
+        array_type: BackingData = BackingData.NumPy,
+        fill_value: float = 0.0,
+    ) -> None:
+
+        if array_type is BackingData.NumPy:
+            import numpy as np
+
+            data = np.full(tuple(dimensions), fill_value)
+
+        elif array_type is BackingData.Zarr:
+            # Zarr is not shipped by default, but we can try to support it
+            import zarr
+
+            data = zarr.full(dimensions, fill_value)
+
+        # give the data array to the viewer.
+        # Replace blank names with None so the Image class generates a name
+        viewer.add_image(
+            name=name if len(name) else None,
+            data=data,
+        )
+
+    # Once called (i.e. "Create"  is clicked), the widget will be destroyed
+    # This means one click of "New image layer" will produce exactly one image,
+    # Which is consistent with the other new layer buttons.
+    _new_image_widget.called.connect(_new_image_widget.close)
+
+    # Show the widget (as a modal dialog, outside of the napari window)
+    _new_image_widget.show()
```

Comment:
a dialog should have set viewer as a parent before the show to inherit viewer style.

Reply:
@Czaki I can't seem to figure out how to do this, what would this look like?

Reply:
`_new_image_widget.native.setParent(viewer.widow.viewer.window._qt_window)` but I write from memory

Reply:
Hmm, that did copy the style but also removed the separate window :confused: 
![image](https://user-images.githubusercontent.com/29754838/171677246-96d602bb-7e13-4e20-950d-33d7e3fde8e3.png)

Reply:
Ok. This is Plain widget, not QDialog instance. 

This should work:
```
from napari.utils.theme import template, theme
from napari.qt import get_stylesheet
from napari.settings import get_settings

theme = get_theme(get_settings().appearance.theme, as_dict=True) _new_image_widget.native.setStyleSheet(template(get_stylesheet(), **theme)
```

But there should be an easier way...